### ISSUE TITLE
Substitution doesn't work if the Key has a character from the start or end values

### ISSUE THREAD

**Issue by EnderBlue** [molinch/ConfigurationSubstitutor#5]
The following test will fail:
```
        [Fact]
        public void Should_get_substituted_value_when_using_long_substituable_pattern()
        {
            var configurationBuilder = new ConfigurationBuilder()
                .AddInMemoryCollection(new Dictionary<string, string>()
                {
                { "Foo", "Hello %(env,Testo)%" },
                { "Testo", "world" }
                })
                .EnableSubstitutions("%(env,", ")%");

            var configuration = configurationBuilder.Build();

            // Act
            var substituted = configuration["Foo"];

            substituted.Should().Be("Hello world");
        }
```

I was able to resolve it by changing the Regex to this:
```
_findSubstitutions = new Regex(@"(?<=" + escapedStart + @")((?!" + escapedStart + ").)((?!" + escapedEnd +").)*(?=" + escapedEnd + @")",
                RegexOptions.Compiled);
```

I am not an expert with regex, so I don't know if this is the best way.  It seems to allow all of the existing tests to pass.

**Comment by molinch** [comment]
@EnderBlue sorry for the delay, I'll try to look into it this evening!

**Comment by aishpriya** [comment]
@molinch any updates on this?

**Comment by molinch** [comment]
Fixed with https://github.com/molinch/ConfigurationSubstitutor/commit/58282dd3524d87ea3c62c0e343a321999da4e382
Published nuget 1.1.2
https://www.nuget.org/packages/ConfigurationSubstitutor/
